### PR TITLE
Update Developer guide for bookmark feature

### LINKED ISSUE #101: Update Developer Guide

### PR DESCRIPTION
**Pull request by lingsihui**
Fixes #101

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -4,6 +4,54 @@
...
+
+API: java.bookmarkRun
+
+The bookmark component consists of six major classes: `BookmarkRun`, `BookmarkStorage`, `BookmarkUi`, 
+`BookmarkCategory`, `BookmarkList` and `BookmarkParser`. 
+As shown in figure 1, `BookmarkRun` is 
+* The main class to be called when the bookmark mode is accessed.
+* The main class to access other bookmark classes.
+
+Given below, figure 2, is the sequence diagram of how the classes interact with each
```
Comment (farice9) [inline docs/DeveloperGuide.md]: I think need to Figure 2 instead of figure 2

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -4,6 +4,54 @@
...
+* The main class to access other bookmark classes.
+
+Given below, figure 2, is the sequence diagram of how the classes interact with each 
+other when bookmark mode is accessed from the main function.
+
+[uml diagram]
+![Bookmark Sequence Diagram](<Images/sequence_bookmark.png>)
+
+The bookmark component has two modes: the main bookmark mode and the category mode. 
+As shown in figure 2, when `BookmarkRun` is called,
```
Comment (farice9) [inline docs/DeveloperGuide.md]: Same thing here

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -4,6 +4,54 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
```
Comment (farice9) [inline docs/DeveloperGuide.md]: Are we keeping this line?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -4,6 +4,54 @@
...
+![BookmarkCommand ClassDiagram](<Images/bookmarkCommand_Class>)
+
+Figure 4 shows a more detailed sequence diagram of how the `BookmarkCategory`, `BookmarkCommand` 
+and `BookmarkList` interacts with each other for the scenario when the user input an `AddLinkCommand` 
+in Bookmark mode.
+
+![BookmarkCommand ClassDiagram](<Images/AddCommand_sequenceDiagram>)
+
+
+
```
Comment (farice9) [inline docs/DeveloperGuide.md:54]: Everything else looks goood

**Review by farice9 — CHANGES_REQUESTED** [review]

**Review by farice9 — APPROVED** [review]

**Comment by farice9** [comment]
Sorry wrongly approved, wait for new push before merging

**Comment by slightlyharp** [comment]
LGTM

**Pull request merged**